### ISSUE TITLE
How can I share some value without sharing Context.Key object instance?

### ISSUE THREAD

**Issue by hojongs** [grpc/grpc-java#7024]
### Summary
How can I store and get a value from Context without sharing Context.Key instance?

### Detail

I want to share some value with grpc Context. but I can't get the stored value with Context.Key instance

I tried this snippet
```kotlin
Context.key("A") == Context.key("A")
```

The result was
 ```
false
```

But I expected 
```
true
```

It means if I want to get value from a Context, I should use same Context.Key instance.
It is strange. Because, I'm trying to share some values with Context and when I get the stored value I should use same Context.Key.
To share some value, should share the key object instance. (not the key value)

For example (with kotlin)

```kotlin
// Function A
val myKey = Context.key("myKey")

Context.current()
    .withValue(key, "myVal")
    .attach()
```

```kotlin
// Function B, After call A
Context.key<String>("myKey").get() // == null // I think, It should be "myVal"

myKey.get() // == "myVal" 
```

If I can share `myKey` object instance, I don't need Context to share value.
What about multi-threaded case? It is more difficult to share the Context.Key object instance.

I can't understand use of Context and Context.key to share value.

### In short, My question is.
How can I store and get a value from Context without sharing Context.Key instance?

**Comment by ericgribkoff** [comment]
You're correct that multiple `Context.Key` can share the same debug name but still won't return true when tested for equality. From the doc at https://grpc.github.io/grpc-java/javadoc/io/grpc/Context.html#key-java.lang.String-:

> Create a Context.Key with the given debug name. Multiple different keys may have the same name; the name is intended for debugging purposes and does not impact behavior.

You need to use the same `Context.Key` reference when inserting and retrieving values from the context. A simple way to achieve this is to create your key as essentially a global constant and use it when setting and retrieving values. I didn't look through it carefully but https://github.com/saturnism/grpc-by-example-java/tree/master/metadata-context-example shows example code that uses `Context.Key` to propagate values.

**Comment by hojongs** [comment]
Thank you for your answer. I was missing the doc of Context.key API
But I think it is confuse even now.
Generally, key's name will be recognized it is identifier by people (In my opinion)
the name parameter of Context.key(name) is debugging name, so Let me give a suggestion.
- Make Context.key(name) function have default parameter value (the null)
  - or overload key function without any parameter.
  - It represents the name parameter is optional.
```java
    Key() {
      this(null, null);
    }
```
```java
    Key(String name = null) {
      this(name, null);
    }
```

**Comment by ericgribkoff** [comment]
Since the question here is resolved, I'm going to mark this issue as closed (discussion on the updates can continue on the PR). Thanks!
